save-backend/src/main/kotlin/com/saveourtool/save/backend/service/ProjectService.kt

```diff
@@ -149,16 +149,33 @@ class ProjectService(
     /**
      * @param projectFilters is filter for [projects]
      * @return project's with filter
-     */
-    fun getFiltered(projectFilters: ProjectFilters): List<Project> =
-            when (projectFilters.organizationName.isBlank() to projectFilters.name.isBlank()) {
-                true to true -> projectRepository.findByStatusIn(projectFilters.statuses)
-                true to false -> projectRepository.findByNameLikeAndStatusIn(wrapValue(projectFilters.name), projectFilters.statuses)
-                false to true -> projectRepository.findByOrganizationNameAndStatusIn(projectFilters.organizationName, projectFilters.statuses)
-                false to false -> findByNameAndOrganizationNameAndStatusIn(projectFilters.name, projectFilters.organizationName, projectFilters.statuses)
-                    ?.let { listOf(it) }.orEmpty()
-                else -> throw IllegalStateException("Impossible state")
-            }
+     * @throws IllegalStateException
+     */
+    fun getFiltered(projectFilters: ProjectFilters): List<Project> = when (projectFilters.organizationName.isBlank() to projectFilters.name.isBlank()) {
+        true to true -> projectRepository.findByStatusIn(projectFilters.statuses)
+        true to false -> projectRepository.findByNameLikeAndStatusIn(
+            wrapValue(projectFilters.name),
+            projectFilters.statuses
+        )
+
+        false to true -> projectRepository.findByOrganizationNameAndStatusIn(
+            projectFilters.organizationName,
+            projectFilters.statuses
+        )
+
+        false to false -> findByNameAndOrganizationNameAndStatusIn(
+            projectFilters.name,
+            projectFilters.organizationName,
+            projectFilters.statuses
+        )
+            ?.let { listOf(it) }.orEmpty()
+
+        else -> throw IllegalStateException("Impossible state")
+    }.filter { project ->
+        projectFilters.public?.let {
+            project.public == it
```

Comment:
can we  support this filtering on backend side?

Reply:
fixed